app/models/task.py

```diff
@@ -2,4 +2,50 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goals = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        if self.completed_at is None:
+            if self.goal_id is None:
+                task_dict = {
+                    "id": self.task_id,
+                    "title": self.title,
+                    "description": self.description,
+                    "is_complete": False
+                }
+            else:
+                task_dict = {
+                    "id": self.task_id,
+                    "goal_id": self.goal_id,
+                    "title": self.title,
+                    "description": self.description,
+                    "is_complete": False
+                }
+        else:
+            task_dict = {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": True
+            }
+        return task_dict
```

Comment:
This method is a little overcomplicated.  Consider maybe making the basic dictionary and then adding key-value pairs for the additional fields in if-else blocks.